library/private/options_tools.h.in

```diff
@@ -181,6 +183,21 @@ ratio_t parse(const std::string& str)
   }
 }
 
+//----------------------------------------------------------------------------
+/**
+ * Parse provided string into a vector3_t.
+ */
+template<>
+vector3_t parse(const std::string& str)
+{
+  std::vector vec = parse<std::vector<double>>(str);
```

Comment:
this is indeed only the naive approach, you also need to add the "+Y" syntax, lets also add the `spherical()` syntax.